leshan-core/src/test/java/org/eclipse/leshan/core/node/codec/LwM2mNodeEncoderTest.java

```diff
@@ -402,14 +408,17 @@ public void senml_json_encode_timestamped_nodes() throws CodecException {
 
     @Test
     public void senml_cbor_encode_timestamped_nodes() throws CodecException {
+        Instant timestamp = Instant.ofEpochSecond(500_000_000);
         TimestampedLwM2mNodes timestampedLwM2mNodes = TimestampedLwM2mNodes.builder()
-                .put(500_000_004L, new LwM2mPath(0, 0, 0), LwM2mSingleResource.newStringResource(0, "SampleString"))
-                .put(500_000_005L, new LwM2mPath(0, 0, 1), LwM2mSingleResource.newBooleanResource(1, false))
-                .put(500_000_006L, new LwM2mPath(0, 0, 2), LwM2mSingleResource.newIntegerResource(2, 456)).build();
+                .put(timestamp.plusSeconds(4), new LwM2mPath(0, 0, 0),
+                        LwM2mSingleResource.newStringResource(0, "SampleString"))
+                .put(timestamp.plusSeconds(5), new LwM2mPath(0, 0, 1), LwM2mSingleResource.newBooleanResource(1, false))
+                .put(timestamp.plusSeconds(6), new LwM2mPath(0, 0, 2), LwM2mSingleResource.newIntegerResource(2, 456))
+                .build();
 
         byte[] encoded = encoder.encodeTimestampedNodes(timestampedLwM2mNodes, ContentFormat.SENML_CBOR, model);
 
-        String expectedString = "83a321662f302f302f30221a1dcd6504036c53616d706c65537472696e67a321662f302f302f31221a1dcd650504f4a321662f302f302f32221a1dcd6506021901c8";
+        String expectedString = "83a321662f302f302f3022c482001a1dcd6504036c53616d706c65537472696e67a321662f302f302f3122c482001a1dcd650504f4a321662f302f302f3222c482001a1dcd6506021901c8";
```

Comment:
I'm not sure to understand why the serialization change here ?
At first sight I would say this is not expected.

Reply:
I believe it's because `BigDecimal` is serialized differently than `long`, as you noticed [here](https://github.com/eclipse/leshan/pull/1313#discussion_r981181837) for JSON.

Reply:
mmhh but here this is not about JSON but CBOR, so ?  (maybe I missed something)

Reply:
`CBORObject` encodes `long` and `BigDecimal` differently, similarly to JSON encoding. Let's look at how it encodes the number 100 for both cases:

- For long
`CBORObject.NewMap().Add(1, 100).EncodeToBytes()` returns `[-95, 1, 24, 100]`

- For BigDecimal
`CBORObject.NewMap().Add(1, BigDecimal.valueOf(100)).EncodeToBytes()` returns `[-95, 1, -60, -126, 0, 24, 100]`

Reply:
Yep you're right. I don't know if this is an issue...
By default the library we used encode Bigdecimal in [CBOR Decimal Fractions and Bigfloats](https://www.rfc-editor.org/rfc/rfc7049#section-2.4.3)

What do you thing about that ?
Considering https://github.com/eclipse/leshan/issues/1304#issuecomment-1260676527, how should we encode this value for our first version ?

Reply:
Using double means that we lose precision beyond 0.1 microsecond for real epoch time in seconds, since the integer part is large enough so that it can't fit all the decimal points:
- double value `1664455172.123456789` is printed out like `1.6644551721234567E9`, so we lose the two digits, 8 and 9 at the end.

Currently however, there aren't very many devices (especially the client ones used in IoT) that can produce such precise timestamps, at least not using Java. As an example:
- `Instant.now().getNano()` would return something like `989921500`
- `System.nanoTime()` would return something like `86050945362600`

As you can see, in both cases I was _de facto_ only able to obtain 7 decimal points (so up to 0.1 microsecond).

Now, I don't know if that's a good enough argument, especially since it's not future-proof. My take is that we should use what we're using right now - whether it be Bigfloat or anything granting more precision than a double as a way to enable true nanosecond precision.

Reply:
Ok let's using this for now.
I will create a dedicated issue about all of this to centralize all idea in one place.